packages/augur-core/src/contracts/gsn/v2/Forwarder.sol

```diff
@@ -2,61 +2,69 @@
 pragma solidity 0.5.15;
 pragma experimental ABIEncoderV2;
 
-import 'ROOT/gsn/v1/ECDSA.sol';
+import "ROOT/gsn/v1/ECDSA.sol";
+import "ROOT/gsn/v2/utils/GsnUtils.sol";
+import "ROOT/gsn/v2/interfaces/IForwarder.sol";
+import "ROOT/gsn/v2/interfaces/ISignatureVerifier.sol";
 
-import "ROOT/gsn/v2/utils/GSNTypes.sol";
-import "ROOT/gsn/v2/interfaces/ITrustedForwarder.sol";
 
+contract Forwarder is IForwarder {
 
-contract TrustedForwarder is ITrustedForwarder {
     using ECDSA for bytes32;
 
-    // Nonces of senders, used to prevent replay attacks
-    mapping(address => uint256) private nonces;
-
     // solhint-disable-next-line max-line-length
-    bytes32 public constant RELAY_REQUEST_TYPEHASH = keccak256("RelayRequest(address target,bytes encodedFunction,GasData gasData,RelayData relayData)GasData(uint256 gasLimit,uint256 gasPrice,uint256 pctRelayFee,uint256 baseRelayFee)RelayData(address senderAddress,uint256 senderNonce,address relayWorker,address paymaster)");
+    bytes32 public constant RELAY_REQUEST_TYPEHASH = keccak256("RelayRequest(address target,bytes encodedFunction,GasData gasData,RelayData relayData)GasData(uint256 gasLimit,uint256 gasPrice,uint256 pctRelayFee,uint256 baseRelayFee)RelayData(address senderAddress,uint256 senderNonce,address relayWorker,address paymaster,address forwarder)");
 
     // solhint-disable-next-line max-line-length
     bytes32 public constant CALLDATA_TYPEHASH = keccak256("GasData(uint256 gasLimit,uint256 gasPrice,uint256 pctRelayFee,uint256 baseRelayFee)");
 
     // solhint-disable-next-line max-line-length
-    bytes32 public constant RELAYDATA_TYPEHASH = keccak256("RelayData(address senderAddress,uint256 senderNonce,address relayWorker,address paymaster)");
+    bytes32 public constant RELAYDATA_TYPEHASH = keccak256("RelayData(address senderAddress,uint256 senderNonce,address relayWorker,address paymaster,address forwarder)");
+
+
+    string public versionSM = "augur-forwarder";
+
+    function versionForwarder() external view returns (string memory) {
+        return "2.0.0-alpha.1+opengsn.forwarder.iforwarder";
```

Comment:
seems to be in multiple places, i assume they won't have to be updated in the future